prow/tide/search.go

```diff
@@ -27,121 +27,50 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type searchExecutor func(start, end time.Time) ([]PullRequest, int /*true match count*/, error)
+type querier func(context.Context, interface{}, map[string]interface{}) error
 
-func newSearchExecutor(ctx context.Context, ghc githubClient, log *logrus.Entry, q string) searchExecutor {
-	return func(start, end time.Time) ([]PullRequest, int, error) {
-		requestStart := time.Now()
-		datedQuery := fmt.Sprintf("%s %s", q, dateToken(start, end))
-		vars := map[string]interface{}{
-			"query":        githubql.String(datedQuery),
-			"searchCursor": (*githubql.String)(nil),
-		}
-		var totalCost, remaining int
-		var totalMatches int
-		var ret []PullRequest
-		for {
-			sq := searchQuery{}
-			if err := ghc.Query(ctx, &sq, vars); err != nil {
-				return nil, 0, fmt.Errorf("error handling query: %q, err: %v", datedQuery, err)
-			}
-			totalCost += int(sq.RateLimit.Cost)
-			remaining = int(sq.RateLimit.Remaining)
-			totalMatches = int(sq.Search.IssueCount)
-			// If the search won't return all results, abort.
-			if totalMatches > 1000 {
-				return nil, totalMatches, nil
-			}
-			for _, n := range sq.Search.Nodes {
-				ret = append(ret, n.PullRequest)
-			}
-			if !sq.Search.PageInfo.HasNextPage {
-				break
-			}
-			vars["searchCursor"] = githubql.NewString(sq.Search.PageInfo.EndCursor)
-		}
-		log.WithFields(logrus.Fields{
-			"query":    datedQuery,
-			"start":    start.String(),
-			"end":      end.String(),
-			"duration": time.Since(requestStart).String(),
-		}).Debugf("Query returned %d PRs and cost %d point(s). %d remaining.", len(ret), totalCost, remaining)
-		return ret, totalMatches, nil
-	}
-}
-
-func (q searchExecutor) search() ([]PullRequest, error) {
-	prs, _, err := q.searchRange(time.Time{}, time.Now())
-	return prs, err
-}
-
-func (q searchExecutor) searchSince(t time.Time) ([]PullRequest, error) {
-	prs, _, err := q.searchRange(t, time.Now())
-	return prs, err
-}
-
-func (q searchExecutor) searchRange(start, end time.Time) ([]PullRequest, int, error) {
-	// Adjust times to be after GitHub was founded to avoid querying empty time
-	// ranges.
+func search(query querier, log *logrus.Entry, q string, start, end time.Time) ([]PullRequest, error) {
 	if start.Before(github.FoundingYear) {
```

Comment:
Note that the minimum start time for searches is this value, which continues to be the case.